### PR TITLE
cli tool

### PR DESCRIPTION
**Pull request by pete965**

### CODE REVIEW

File: incubator-plugins/cli-tool/src/main/java/CommandLineToolMain.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        EasyEcsSDK.setProperty(ConfigKeys.ALIYUN_EASY_ECS_SDK_ACCESS_SECRET, secretKey);
+        EasyEcsSDK.init();
+    }
+
+    private static void reportRecommendations(Response<List<PreemptiveInstanceRecommendation>> response) {
+        if (CollectionUtils.isEmpty(response.getData())) {
+            System.out.println("Blank Recommendation Response");
+            return;
+        }
+        System.out.println(JSON.toJSONString(response));
```
Comment (DavyJones2010) [inline incubator-plugins/cli-tool/src/main/java/CommandLineToolMain.java:245]: duplicate output

File: incubator-plugins/cli-tool/src/main/java/Constant.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public static final String INSTANCETYPE_OPTION_SHORT = "i";
+    public static final String INSTANCETYPE_OPTION_LONG = "instanceType";
+    public static final String LIMIT_OPTION_SHORT = "l";
+    public static final String LIMIT_OPTION_LONG = "limit";
+    public static final String CATEGORY_OPTION_SHORT = "p";
+    public static final String CATEGORY_OPTION_LONG = "productCategory";
+    public static final String STRATEGY_OPTION_SHORT = "st";
+    public static final String STRATEGY_OPTION_LONG = "strategy";
+    public static final String FILE_OPTION_SHORT = "f";
+    public static final String FILE_OPTION_LONG = "file";
```
Comment (DavyJones2010) [inline incubator-plugins/cli-tool/src/main/java/Constant.java:44]: --profile

File: incubator-plugins/cli-tool/src/main/java/Constant.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+/**
+ * @author xianzhi
+ */
+public class Constant {
+    public static final String HELP_OPTION_SHORT = "h";
+    public static final String HELP_OPTION_LONG = "help";
+    public static final String AK_OPTION_SHORT = "a";
+    public static final String AK_OPTION_LONG = "ak";
```
Comment (DavyJones2010) [inline incubator-plugins/cli-tool/src/main/java/Constant.java:24]: see
https://help.aliyun.com/document_detail/121198.html?spm=a2c4g.11186623.6.552.dc0b57b8sDLxWz

File: incubator-plugins/cli-tool/src/test/java/testCommandLineTool.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        commandLine = getCommandLine(argList);
+        Assert.assertNotNull(commandLine);
+        regions = CommandLineToolMain.getRegions(commandLine);
+        Assert.assertNotNull(regions);
+        Assert.assertEquals(1, regions.size());
+
+    }
+
+    @Test
+    public void testGetUserCredential() {
```
Comment (DavyJones2010) [inline incubator-plugins/cli-tool/src/test/java/testCommandLineTool.java:117]: delete unused test

File: incubator-plugins/cli-tool/pom.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        <artifactId>incubator-plugins</artifactId>
+        <groupId>com.aliyun.ecs.easysdk</groupId>
+        <version>1.1.0</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <dependencies>
+        <dependency>
+            <groupId>commons-cli</groupId>
+            <artifactId>commons-cli</artifactId>
```
[the remark this answers is not in the document]
Comment (pete965) [reply incubator-plugins/cli-tool/pom.xml:15]: Done

File: incubator-plugins/cli-tool/pom.xml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>incubator-plugins</artifactId>
+        <groupId>com.aliyun.ecs.easysdk</groupId>
```
[the remark this answers is not in the document]
Comment (pete965) [reply incubator-plugins/cli-tool/pom.xml:7]: Done

File: incubator-plugins/cli-tool/pom.xml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            <artifactId>preemptive-instance-recommendation</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+    </dependencies>
+
+    <artifactId>cli-tool</artifactId>
```
[the remark this answers is not in the document]
Comment (pete965) [reply incubator-plugins/cli-tool/pom.xml:32]: Done

**Pull request merged**